### PR TITLE
Added Sentry Logs | Added Component analysis Threshold checks.

### PR DESCRIPTION
**Pull request by deepak1725**

### CODE REVIEW

File: src/api.py [hunk 1]
```diff
@@ -137,11 +141,17 @@ def metrics_colletion():
...
     try:
-        input_json = request.get_json()
         assert all(inputs in input_json for inputs in payload)
         assert type(input_json['value']) == float
         value = input_json['value']
+        endpoint = input_json['endpoint']
+
+        if endpoint == 'api_v1.get_component_analysis' and value > 2:
+            # In Component analysis if value is more than 2secs, raise error
+            raise Exception("Component Analysis exceeds threshold value")
```
Comment (miteshvp) [inline src/api.py]: you should log the error and not raise the exception

File: src/api.py [hunk 2]
```diff
@@ -137,11 +141,17 @@ def metrics_colletion():
...
     payload = ['endpoint', 'value', 'request_method', 'status_code', 'pid']
+    input_json = request.get_json()
     try:
-        input_json = request.get_json()
         assert all(inputs in input_json for inputs in payload)
         assert type(input_json['value']) == float
         value = input_json['value']
+        endpoint = input_json['endpoint']
+
+        if endpoint == 'api_v1.get_component_analysis' and value > 2:
```
Comment (miteshvp) [inline src/api.py:151]: typo in endpoint name

**Review by miteshvp — CHANGES_REQUESTED** [review]

File: openshift/template.yaml [hunk 3]
```diff
@@ -48,6 +48,11 @@ objects:
             value: "${CLASS_TYPE}"
           - name: FLASK_LOGGING_LEVEL
             value: ${FLASK_LOGGING_LEVEL}
+          - name: SENTRY_DSN
```
Comment (miteshvp) [inline openshift/template.yaml:51]: Do we have this secret already in stage and prod?

**Review by miteshvp — APPROVED** [review]
Just one question. Otherwise LGTM. If that secret is not created, it will not pick anything for sentry.

File: openshift/template.yaml (hunk 3 above)
Comment (deepak1725) [reply openshift/template.yaml:51]: If secrects are in cluster namespace then Yes, This key is used for `bayesian-api` also.
If Secrects are for individual services, then No

**Pull request merged**